mm2src/lp_stats.rs

```diff
@@ -0,0 +1,285 @@
+
+    spawn(stat_collection_loop(ctx, interval));
+
+    let res = json!({
+        "result": "success"
+    });
+    Response::builder()
+        .body(json::to_vec(&res).expect("Serialization failed"))
+        .map_err(|e| ERRL!("{}", e))
+}
+
+#[cfg(not(target_arch = "wasm32"))]
+async fn stat_collection_loop(ctx: MmArc, interval: f64) {
+    use crate::mm2::database::stats_nodes::{select_peers_addresses, select_peers_names};
+
+    loop {
+        if ctx.is_stopping() {
+            break;
+        };
+        {
+            let peers_addresses = match select_peers_addresses(&ctx) {
+                Ok(p) => p,
+                Err(e) => {
+                    log::error!("Error selecting peers addresses from db: {}", e);
+                    Timer::sleep(10.).await;
+                    continue;
+                },
+            };
+
+            let peers_names = match select_peers_names(&ctx) {
+                Ok(n) => n,
+                Err(e) => {
+                    log::error!("Error selecting peers names from db: {}", e);
+                    Timer::sleep(10.).await;
+                    continue;
+                },
+            };
+
+            let timestamp = now_ms() / 1000;
+            let get_versions_res = match request_addresses::<String>(
+                ctx.clone(),
+                P2PRequest::NetworkInfo(NetworkInfoRequest::GetMm2Version),
+                peers_addresses,
+            )
+            .await
+            {
+                Ok(res) => res,
+                Err(e) => {
+                    log::error!("Error getting nodes versions from peers: {}", e);
+                    Timer::sleep(10.).await;
+                    continue;
+                },
+            };
+
+            for (peer_id, response) in get_versions_res {
+                let name = match peers_names.get(&peer_id.to_string()) {
+                    Some(n) => n.clone(),
+                    None => continue,
+                };
+
+                match response {
+                    PeerDecodedResponse::Ok(version) => {
+                        let node_version_stat = NodeVersionStat {
+                            name,
+                            version,
+                            timestamp,
+                        };
+                        if let Err(e) = insert_node_version_stat_to_db(&ctx, &node_version_stat) {
+                            log::error!("Error inserting nodes versions into db: {}", e);
+                            continue;
+                        };
+                    },
+                    // If a node returns an error or no response it will not be added to the stats table
+                    // A simple count for every node in db will return a count for the number of responses recieved
+                    PeerDecodedResponse::Err(e) => {
+                        log::error!("Node {} responded to version request with error: {}", name, e);
```

Comment:
I think it's worth saving the errors to the DB too: it's easy to lose MM logs in a long timeframe so it will be hard to understand why a specific node misses some records.